### PR TITLE
Add reaction annotation tests

### PR DESCRIPTION
**Pull request by vm-vh**
Updated src/memote/reactions.jl and test/memote.jl to include reaction annotation tests. (Tests written together with Flora @Fl-Sch but we figured only doing 1 PR would make more sense)

### CODE REVIEW

**Comment by stelmo** [comment]
nice! I will look at it tomorrow

File: src/memote/reactions.jl [hunk 1]
```diff
@@ -4,3 +4,88 @@ M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
...
+        "brenda" =>
+            r"^\d+\.-\.-\.-|\d+\.\d+\.-\.-|\d+\.\d+\.\d+\.-|\d+\.\d+\.\d+\.(n)?\d+$",
+        "biocyc" => r"^[A-Z-0-9]+(?<!CHEBI)(\:)?[A-Za-z0-9+_.%-]+$",
+    ),
+)
+    no_annos = Dict{String,Vector{String}}()
+    for rid in reactions(model)
+        annos = reaction_annotations(model, rid)
+        for anno_kw in keys(REACTION_ANNOTATIONS)
+            if haskey(annos, anno_kw) == true
```
Comment (exaexa) [inline src/memote/reactions.jl:83]: `== true`.

File: src/memote/reactions.jl [hunk 2]
```diff
@@ -4,3 +4,88 @@ M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
...
+                !in(anno_kw, keys(no_annos)) && (no_annos[anno_kw] = [])
+                in.(nothing, Ref(match.(REACTION_ANNOTATIONS[anno_kw], annos[anno_kw]))) &&
+                    push!(no_annos[anno_kw], rid)
+            end
+        end
+    end
+    return no_annos
```
Comment (exaexa) [inline src/memote/reactions.jl:90]:
```suggestion
    return Dict(
        rid => [anno_kw
          for anno_kw=keys(REACTION_ANNOTATIONS)
          if haskey(reaction_annotations(model, rid), anno_kw)
          # ... etc
        ] for rid=reactions(model)
    )
```

Also, this:
```julia
                !in(anno_kw, keys(no_annos)) && (no_annos[anno_kw] = [])
                in.(nothing, Ref(match.(REACTION_ANNOTATIONS[anno_kw], annos[anno_kw]))) &&
                    push!(no_annos[anno_kw], rid)
```
is sometimes resetting and sometimes pushing into anno_kw, effectively likely ending up with a random Dict-order dependent erasure of no_annos. Which is likely wrong.

File: src/memote/reactions.jl [hunk 3]
```diff
@@ -4,3 +4,88 @@ M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
...
+)
+    missing_annos = Dict{String,Vector{String}}()
+    for keyword in annotation_keywords
+        missing_annos[keyword] = []
+        for reaction_id in reactions(model)
+            reaction_annos = reaction_annotations(model, reaction_id)
+            !haskey(reaction_annos, keyword) && push!(missing_annos[keyword], reaction_id)
+        end
+    end
+    return missing_annos
```
Comment (exaexa) [inline src/memote/reactions.jl:52]:
```suggestion
    return Dict(
        keyword => [reaction_id
            for reaction_id=reactions(model)
            if !haskey(reaction_annotations(model, reaction_id), keyword) ]
        for keyword=annotation_keywords
    )
```

File: src/memote/reactions.jl (hunk 2 above)
Comment (exaexa) [reply src/memote/reactions.jl:90]: PS why is that constant written in SCREAMING_TOTAL_SNAKE_CASE, and not put safely in the config?

**Review by exaexa — CHANGES_REQUESTED** [review]
there's likely some randomness introduced by going thru REACTION_ANNOTATIONS, needs fixing

File: src/memote/reactions.jl (hunk 2 above)
Comment (stelmo) [reply src/memote/reactions.jl:90]: I changed this in #37 to use `get!`

**Comment by exaexa** [comment]
Pls keep open until merged elsewhere :D

**Comment by exaexa** [comment]
(merged as a part of #37)

**Pull request merged**